packages/tx/src/baseTransaction.ts

```diff
@@ -0,0 +1,193 @@
+import Common from '@ethereumjs/common'
+import {
+  Address,
+  BN,
+  toBuffer,
+  MAX_INTEGER,
+  unpadBuffer,
+  ecsign,
+  publicToAddress,
+} from 'ethereumjs-util'
+import { BaseTransactionData, BaseTxOptions, DEFAULT_COMMON, JsonTx } from './types'
+
+export abstract class BaseTransaction<TransactionObject> {
+  public readonly nonce: BN
+  public readonly gasLimit: BN
+  public readonly gasPrice: BN
+  public readonly to?: Address
+  public readonly value: BN
+  public readonly data: Buffer
+  public readonly common: Common
+
+  constructor(txData: BaseTransactionData, txOptions: BaseTxOptions = {}) {
+    const { nonce, gasLimit, gasPrice, to, value, data } = txData
+
+    this.nonce = new BN(toBuffer(nonce))
+    this.gasPrice = new BN(toBuffer(gasPrice))
+    this.gasLimit = new BN(toBuffer(gasLimit))
+    this.to = to ? new Address(toBuffer(to)) : undefined
+    this.value = new BN(toBuffer(value))
+    this.data = toBuffer(data)
+
+    const validateCannotExceedMaxInteger = {
+      nonce: this.nonce,
+      gasPrice: this.gasPrice,
+      gasLimit: this.gasLimit,
+      value: this.value,
+    }
+
+    this.validateExcdeedsMaxInteger(validateCannotExceedMaxInteger)
+
+    this.common =
+      (txOptions.common &&
+        Object.assign(Object.create(Object.getPrototypeOf(txOptions.common)), txOptions.common)) ??
+      DEFAULT_COMMON
```

Comment:
`DEFAULT_COMMON` won't work atm for the `EIP2930Transaction` without the activation of the EIPs. Not sure, do we need to separate the Common objects here and instantiate one with `eips: [2718, 2930]` or can we use the one with the EIP activations also for the `LegacyTransaction`? Or will this have side-effects? 🤔

Reply:
Ah, likely best to directly set to `berlin` I guess.

Reply:
If we set default common to Berlin, then these EIPs are activated by default right?

Also a side note: I think we should move `DEFAULT_COMMON` to Common, so we can have package-wide "default" common objects. I also think it would be helpful if we add a `copy` method to Common.

Reply:
> I also think it would be helpful if we add a `copy` method to Common.

that's a great idea

Reply:
Have been thinking about this a bit, this brings again up the compatibility question. So I guess we can't set to `berlin` yet, the VM uses the default blocks/txs and this will therefore get in an inconsistent state if the VM and the other libs are still on `istanbul`.

Might be a bit counterintuitive on first thought, but I guess the technically clean way to do it is to keep default on plain `istanbul` - even without the EIPs, since this is also a switch on the rule set - and then just let this throw on default when an `EIP2930Transaction` is instantiated without an explicit `common`. I think this makes it clear for everyone and should be acceptable if properly communicated (in the constructor docs).

Together with the `london` implementation the time then might be there for another major release series and we then switch directly to `london`.

WDYGT? 🤔

`DEFAULT_COMMON` to Common: that's an interesting idea, we should be careful with this though and make sure it is properly thought through and we don't introduce any redundancy - potential source for inconsistencies here - we already have the `DEFAULT_HARDFORK` there. We might then also want to do `DEFAULT_CHAIN` set to `mainnet` (likely?). Not 100% sure but supportive on first thought, likely some case for a separate issue though.

Reply:
I think I agree here - I don't think it makes sense to default it to Berlin, especially since it is not yet activated on mainnet. It would be very weird to default the VM to currently non-activated new logic.

About creating a `DEFAULT_COMMON` in Common: this is not in the scope of this PR I'd say, we can open an issue for it and then discuss this. I will open an issue.

Reply:
Issue #1128 

Again, it would make sense that we default to Istanbul. It is therefore not possible to create `EIP2930Transaction` if you don't pass an explicit Common to it with the EIPs activated.